datalake/src/main/java/com/sequenceiq/datalake/service/upgrade/database/SdxDatabaseServerUpgradeService.java

```diff
@@ -85,7 +85,7 @@ public SdxUpgradeDatabaseServerResponse upgrade(NameOrCrn sdxNameOrCrn, TargetMa
         if (!isUpgradeNeeded(targetMajorVersion, cluster)) {
             throwAlreadyOnLatestError(cluster, targetMajorVersion);
         }
-
+        cloudbreakStackService.checkUpgradeRdsByClusterNameInternal(cluster, targetMajorVersion);
```

Comment:
I would extract it to a method named
"validateNoAttachedDataHubsAreRunning"

Reply:
As it is an internal Api call which can be extended with more validation steps, I wouldn't change the method name.